app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/event_screen/EventViewModel.kt

```diff
@@ -472,8 +482,39 @@ class EventViewModel @Inject constructor(
                     }
                 }
             }
-            is JoinEvent -> {}   // todo implement
-            is LeaveEvent -> {}  // todo implement
+            is LeaveEvent -> {
+                _state.update { _state ->
+                    _state.copy(
+                        event = _state.event?.copy(
+                            attendees = _state.event.attendees.map {
+                                if (it.id == state.value.authInfo?.userId) {
+                                    it.copy(isGoing = false)
+                                } else {
+                                    it
+                                }
+                            },
+                            isGoing = false
+                        )
+                    )
+                }
+            }
+            is JoinEvent -> {
+                _state.update { _state ->
+                    _state.copy(
+                        event = _state.event?.copy(
+                            attendees = _state.event.attendees.map {
+                                if (it.id == state.value.authInfo?.userId) {
+                                    it.copy(isGoing = true)
+                                } else {
+                                    it
+                                }
+                            },
+                            isGoing = true
+                        )
+
+                    )
+                }
+            }
```

Comment:
Could make sense to create a function `setIsGoingForLocalUser(isGoing)` to avoid this duplication 😄

Reply:
ok, will refactor this.